### ISSUE TITLE
process.sendToMain() bug

### ISSUE THREAD

**Issue by enricogior** [thaliproject/jxcore#66]
The following script shows the bug in the `process.sendToMain()` method.
```
jxcore.tasks.setThreadCount(2);

var count = 0;

function runTest() {
  // make sure there are no tasks still running
  jxcore.tasks.killThread(0)
  jxcore.tasks.killThread(1)

  count++;
  if (count > 5) return;

  console.log("---- Count " + count + " -----");
 
  var subTask = function (param) {
      console.log("sending msg to main");
      process.sendToMain({result: "msg " + param.count});
      return 'ok';
  };
   
  jxcore.tasks.on("message", function(threadId, obj) {
        console.log("thread " + threadId + ": " + obj.result);
      });

  jxcore.tasks.addTask(subTask, {count: count}, function (err, result) {
        console.log("subTask done");
        runTest();
      });
}

runTest();
```
The espected result should be:
```
---- Count 1 -----
sending msg to main
thread 0: msg 1
subTask done
---- Count 2 -----
sending msg to main
thread 0: msg 2
subTask done
---- Count 3 -----
sending msg to main
thread 0: msg 3
subTask done
---- Count 4 -----
sending msg to main
thread 0: msg 4
subTask done
```
Instead the result is:
```
---- Count 1 -----
sending msg to main
thread 0: msg 1
subTask done
---- Count 2 -----
sending msg to main
thread 0: msg 2
thread 0: msg 2   <---- unexpected
subTask done
---- Count 3 -----
sending msg to main
thread 0: msg 3
thread 0: msg 3   <---- unexpected
thread 0: msg 3   <---- unexpected
subTask done
---- Count 4 -----
sending msg to main
thread 0: msg 4
thread 0: msg 4   <---- unexpected
thread 0: msg 4   <---- unexpected
thread 0: msg 4   <---- unexpected
subTask done
```

**Comment by enricogior** [comment]
I also verified using the 'emptyQueue' event that each new task is added only after the queue is emptied, and I got the same unexpected results.

**Comment by yaronyg** [comment]
We aren't going to use the tasks api for Thali so we should just work around this bug.
